### PR TITLE
fix: enable all tests

### PR DESCRIPTION
**Pull request by klkvr**
it seems that we had incorrect `cargo nextest` args which caused tests under `./tests` to never run

### CODE REVIEW

**Review by mattsse — APPROVED** [review]

**Comment by klkvr** [comment]
CI failure related to https://github.com/foundry-rs/foundry/issues/7797

Addressed this by rewriting path resolution logic for artifacts. Changes:
- We now just sort files by number of components when resolving paths instead of finding top-most
- We now keep track of a single `taken_paths_lowercase` instead of keeping contracts in groups keyed by `MappedContract`.
- Deduplicated logic for artifact paths creation for contracts and standalone sources by extracing most of the logic into `ArtifactOutput::get_artifact_path`

cc @mattsse

**Comment by klkvr** [comment]
A lot of WIndows tests were outdated, updated those. Will address clippy failure in #115

@mattsse ptal, added some changes using `slash_paths` here and there

File: src/cache.rs [hunk 1]
```diff
@@ -57,7 +57,14 @@ impl<S: CompilerSettings> CompilerCache<S> {
 
     /// Removes entry for the given file
     pub fn remove(&mut self, file: &Path) -> Option<CacheEntry<S>> {
-        self.files.remove(file)
+        if let Some(entry) = self.files.remove(file) {
```
Comment (DaniPopes) [inline src/cache.rs:60]: `.inspect(|entry| { ... })`

File: src/config.rs [hunk 2]
```diff
@@ -67,6 +67,15 @@ impl ProjectPathsConfig {
         let graph = Graph::resolve_sources(self, sources)?;
         let ordered_deps = collect_ordered_deps(&flatten_target, self, &graph)?;
 
+        #[cfg(windows)]
+        let ordered_deps = {
+            use path_slash::PathBufExt;
+            ordered_deps
```
Comment (DaniPopes) [inline src/config.rs:73]: `for p in &mut ordered_deps { *p = ... }`?
or since we only iterate it doesn't need collect

File: src/flatten.rs [hunk 3]
```diff
@@ -191,6 +191,15 @@ impl Flattener {
 
         let ordered_sources = collect_ordered_deps(&target.to_path_buf(), &project.paths, &graph)?;
 
+        #[cfg(windows)]
+        let ordered_sources = {
+            use path_slash::PathBufExt;
+            ordered_sources
```
Comment (DaniPopes) [inline src/flatten.rs:197]: same as above

**Review by DaniPopes — CHANGES_REQUESTED** [review]
lgtm, pending clippy and @mattsse

File: src/cache.rs (hunk 1 above)
Comment (klkvr) [reply src/cache.rs]: https://github.com/foundry-rs/compilers/actions/runs/8923860095/job/24508985313?pr=118#step:5:571

should I bump MSRV?

Comment (DaniPopes) [reply src/cache.rs]: Ah no then use `map`

**Comment by klkvr** [comment]
Updated tests to use `fd_lock` for locking installer, should be less flaky now

**Review by mattsse — APPROVED** [review]
ty!

**Pull request merged**